format/spec.md

```diff
@@ -454,35 +457,40 @@ The schema of a manifest file is a struct called `manifest_entry` with the follo
+| _optional_ | _optional_ | _optional_ | **`109  value_counts`**           | `map<119: int, 120: long>`   | Map from column id to number of values in the column (including null and NaN values) |
+| _optional_ | _optional_ | _optional_ | **`110  null_value_counts`**      | `map<121: int, 122: long>`   | Map from column id to number of null values in the column |
+| _optional_ | _optional_ | _optional_ | **`137  nan_value_counts`**       | `map<138: int, 139: long>`   | Map from column id to number of NaN values in the column |
+| _optional_ | _optional_ | _optional_ | **`111  distinct_counts`**        | `map<123: int, 124: long>`   | Map from column id to number of distinct values in the column; distinct counts must be derived using values in the file by counting or using sketches, but not using methods like merging existing distinct counts |
+| _optional_ | _optional_ | _optional_ | **`125  lower_bounds`**           | `map<126: int, 127: binary>` | Map from column id to lower bound in the column serialized as binary [1]. Each value must be less than or equal to all non-null, non-NaN values in the column for the file [2] |
+| _optional_ | _optional_ | _optional_ | **`128  upper_bounds`**           | `map<129: int, 130: binary>` | Map from column id to upper bound in the column serialized as binary [1]. Each value must be greater than or equal to all non-null, non-Nan values in the column for the file [2] |
+| _optional_ | _optional_ | _optional_ | **`131  key_metadata`**           | `binary`                     | Implementation-specific key metadata for encryption |
+| _optional_ | _optional_ | _optional_ | **`132  split_offsets`**          | `list<133: long>`            | Split offsets for the data file. For example, all row group offsets in a Parquet file. Must be sorted ascending |
+|            | _optional_ | _optional_ | **`135  equality_ids`**           | `list<136: int>`             | Field ids used to determine row equality in equality delete files. Required when `content=2` and should be null otherwise. Fields with ids listed in this column must be present in the delete file |
+| _optional_ | _optional_ | _optional_ | **`140  sort_order_id`**          | `int`                        | ID representing sort order for this file [3]. |
+|            | _optional_ | _optional_ | **`143  referenced_data_file`**   | `string`                     | Fully qualified location (URI with FS scheme) of a data file that all deletes reference [4] |
+|            |            | _optional_ | **`144  blob_offset`**            | `long`                       | The offset in the file where the content starts [5] |
+|            |            | _optional_ | **`145  blob_size_in_bytes`**     | `long`                       | The length of a referenced blob stored in the file [5] |
```

Comment:
These last 3 lines are the only changes. Otherwise the rows are the same but with the v2 requirements copied to the new v3 column.

Reply:
Is there a reason blob_size_in_bytes needed?  Isn't blob_offset sufficient? If this is an optimization to bypass reading the footer, then it seems that compression would also be required here, or explicitly disallowed for the bitmaps?

Reply:
Also, line 464 is a change.

Reply:
We need `blob_size_in_byte` so that we can compare the total Puffin size to the size of the individual DV blob without opening up the Puffin file. It will be useful for maintenance and to generate accurate snapshot stats.

Reply:
I also think the answer to my question above is that blob's referenced here are always assumed to be uncompressed.